control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -3480,3 +3491,237 @@ func healthCheckIdentityProvider(ctx context.Context, hcp *hyperv1.HostedControl
+			AWS: &hyperv1.AWSPlatformStatus{
+				DefaultWorkerSecurityGroupID: sgID,
+			},
+		}
+	}
+	meta.SetStatusCondition(&hcp.Status.Conditions, *condition)
+
+	if err := r.Client.Status().Patch(ctx, hcp, client.MergeFromWithOptions(originalHCP, client.MergeFromWithOptimisticLock{})); err != nil {
+		return fmt.Errorf("failed to update status: %w", err)
+	}
+
+	return creationErr
+}
+
+func awsSecurityGroupFilters(infraID string) []*ec2.Filter {
+	return []*ec2.Filter{
+		{
+			Name:   awssdk.String(fmt.Sprintf("tag:kubernetes.io/cluster/%s", infraID)),
+			Values: []*string{awssdk.String("owned")},
+		},
+		{
+			Name:   awssdk.String("tag:Name"),
+			Values: []*string{awssdk.String(awsSecurityGroupName(infraID))},
+		},
+	}
+}
+
+func awsSecurityGroupName(infraID string) string {
+	return fmt.Sprintf("%s-default-sg", infraID)
+}
+
+func createAWSDefaultSecurityGroup(ctx context.Context, ec2Client ec2iface.EC2API, infraID, vpcID string) (string, error) {
+	logger := ctrl.LoggerFrom(ctx)
+
+	// Determine VPC cidr
+	vpcResult, err := ec2Client.DescribeVpcsWithContext(ctx, &ec2.DescribeVpcsInput{
+		VpcIds: []*string{awssdk.String(vpcID)},
+	})
+	if err != nil {
+		logger.Error(err, "Failed to describe vpc", "vpcID", vpcID)
+		return "", fmt.Errorf("failed to describe vpc %s, code %s", vpcID, awsErrorCode(err))
+	}
+	if len(vpcResult.Vpcs) == 0 {
+		return "", fmt.Errorf("vpc %s not found", vpcID)
+	}
+	vpcCIDR := awssdk.StringValue(vpcResult.Vpcs[0].CidrBlock)
+	describeSGResult, err := ec2Client.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{Filters: awsSecurityGroupFilters(infraID)})
+	if err != nil {
+		logger.Error(err, "Failed to list security groups")
+		return "", fmt.Errorf("cannot list security groups, code: %s", awsErrorCode(err))
+	}
+	sgID := ""
+	var sg *ec2.SecurityGroup
+	if len(describeSGResult.SecurityGroups) > 0 {
+		sg = describeSGResult.SecurityGroups[0]
+		sgID = awssdk.StringValue(sg.GroupId)
+	}
+	if sgID == "" {
+		// Create a security group if one is not found
+		tags := []*ec2.Tag{
+			{
+				Key:   awssdk.String(fmt.Sprintf("kubernetes.io/cluster/%s", infraID)),
+				Value: awssdk.String("owned"),
+			},
+			{
+				Key:   awssdk.String("Name"),
+				Value: awssdk.String(awsSecurityGroupName(infraID)),
+			},
+		}
+		createSGResult, err := ec2Client.CreateSecurityGroup(&ec2.CreateSecurityGroupInput{
+			GroupName:   awssdk.String(awsSecurityGroupName(infraID)),
+			Description: awssdk.String("default worker security group"),
+			VpcId:       awssdk.String(vpcID),
+			TagSpecifications: []*ec2.TagSpecification{
+				{
+					ResourceType: awssdk.String("security-group"),
+					Tags:         tags,
+				},
+			},
+		})
+		if err != nil {
+			logger.Error(err, "Failed to create security group")
+			return "", fmt.Errorf("failed to create security group, code: %s", awsErrorCode(err))
+		}
+		sgID = awssdk.StringValue(createSGResult.GroupId)
+
+		// Fetch just-created SG
+		backoff := wait.Backoff{
```

Comment:
Should be able to use waiters provided by aws-sdk-go here https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/making-requests.html#using-waiters as a suggestion